tests/sentry/hybridcloud/models/test_outbox.py

```diff
@@ -305,6 +306,183 @@ def test_drain_shard_flush_all__concurrent_processing__cooperation(
         assert mock_process_region_outbox.call_count == 2
 
 
+class OutboxDrainReservationTest(TransactionTestCase):
```

Comment:
These tests are a copy/paste of `OutboxDrainTest`, with minor modifications to set the option for the new code path.